hexa/pipelines/views.py

```diff
@@ -87,10 +94,79 @@ def credentials(request: HttpRequest) -> HttpResponse:
     )
 
 
+@require_POST
+@csrf_exempt
+def run_pipeline(
+    request: HttpRequest, id: uuid.UUID, version_number: int = None
+) -> HttpResponse:
+    """Runs a pipeline. The endpoint accepts both form data and JSON payloads.
+
+    To be runnable, the pipeline must be set as public with `has_public_webhook`.
+    Returns:
+        HttpResponse: Returns a dict with the `run_id` key containing the ID of the created run.
+    """
+
+    try:
+        pipeline = Pipeline.objects.get(id=id)
+    except Pipeline.DoesNotExist:
+        return JsonResponse({"error": "Pipeline not found"}, status=404)
+
+    # Only allow pipelines with public webhooks to be run with this endpoint
+    if pipeline.has_public_webhook is False:
+        return JsonResponse({"error": "Pipeline has no public webhook"}, status=400)
+
+    # Get the pipeline version
+    try:
+        pipeline_version = pipeline.last_version
+        if version_number is not None:
+            pipeline_version = PipelineVersion.objects.get(
+                pipeline=pipeline, number=version_number
+            )
+
+        if pipeline_version is None:
+            return JsonResponse({"error": "Pipeline has no version"}, status=400)
+    except PipelineVersion.DoesNotExist:
+        return JsonResponse({"error": "Pipeline version not found"}, status=404)
+
+    # Get the data from the request
+    content_type = request.META.get("CONTENT_TYPE")
+    send_mail_notifications = False
+    config = {}
+    if content_type == "application/x-www-form-urlencoded":
+        send_mail_notifications = request.POST.get("send_mail_notifications", False)
```

Comment:
Shouldn't this be `request.GET.get()`?

Reply:
Nope, We have to do it on GET to have the query params